WordPress/Classes/ViewRelated/Menus/Views/MenusSelectionView.m

```diff
@@ -0,0 +1,271 @@
+            matchingItem = item;
+            break;
+        }
+    }
+    return matchingItem;
+}
+
+- (void)setSelectionItemsExpanded:(BOOL)selectionItemsExpanded
+{
+    if (_selectionItemsExpanded != selectionItemsExpanded) {
+        _selectionItemsExpanded = selectionItemsExpanded;
+        for (MenusSelectionItemView *itemView in self.itemViews) {
+            itemView.hidden = !selectionItemsExpanded;
+            itemView.alpha = itemView.hidden ? 0.0 : 1.0;
+        }
+        
+        self.detailView.showsDesignActive = selectionItemsExpanded;
+    }
+}
+
+- (void)setSelectionItemsExpanded:(BOOL)selectionItemsExpanded animated:(BOOL)animated
+{
+    if (!animated) {
+        self.selectionItemsExpanded = selectionItemsExpanded;
+        return;
+    }
+    [UIView animateWithDuration:0.25 delay:0.0 options:UIViewAnimationOptionCurveEaseOut animations:^{
+        self.selectionItemsExpanded = selectionItemsExpanded;
+    } completion:nil];
+}
+
+#pragma mark - private
+
+- (MenusSelectionItemView *)insertSelectionItemViewWithItem:(MenusSelectionItem *)item
+{
+    MenusSelectionItemView *itemView = [[MenusSelectionItemView alloc] init];
+    itemView.item = item;
+    itemView.delegate = self;
+    
+    NSLayoutConstraint *heightContrainst = [itemView.heightAnchor constraintEqualToConstant:44];
+    heightContrainst.priority = UILayoutPriorityDefaultHigh;
+    heightContrainst.active = YES;
+    itemView.hidden = YES;
+    
+    [self.itemViews addObject:itemView];
+    [self.stackView addArrangedSubview:itemView];
+    
+    // set the width/trailing anchor equal to the stackView
+    [itemView.trailingAnchor constraintEqualToAnchor:self.stackView.trailingAnchor].active = YES;
+    
+    // setup ordering to help with any drawing
+    MenusSelectionItemView *lastItemView = nil;
+    for (MenusSelectionItemView *itemView in self.itemViews) {
+        lastItemView.nextItemView = itemView;
+        itemView.previousItemView = lastItemView;
+        lastItemView = itemView;
+    }
+    
+    return itemView;
+}
+
+- (MenusSelectionItemView *)itemViewForSelectionItem:(MenusSelectionItem *)item
+{
+    MenusSelectionItemView *itemView = nil;
+    for (MenusSelectionItemView *view in self.itemViews) {
+        if (view.item == item) {
+            itemView = view;
+            break;
+        }
+    }
+    return itemView;
+}
+
+- (void)updateDetailsView
+{
+    if (self.selectedItem) {
+        [self.detailView updatewithAvailableItems:self.items.count selectedItem:self.selectedItem];
+    }
+}
+
+#pragma mark - drawing
+
+- (void)setDrawsHighlighted:(BOOL)drawsHighlighted
+{
+    if (_drawsHighlighted != drawsHighlighted) {
+        _drawsHighlighted = drawsHighlighted;
+        self.backgroundColor = drawsHighlighted ? [UIColor colorWithRed:0.99 green:0.99 blue:1.0 alpha:1.0] : [UIColor whiteColor];
+    }
+}
+
+#pragma mark - delegate helpers
+
+- (void)tellDelegateUserInteractionDetectedForTogglingExpansion
```

Comment:
Same comment as in the method in the delegate... feels very verbose (and can probably be simplified).